core/store/src/cold_storage.rs

```diff
@@ -22,6 +24,28 @@ struct StoreWithCache<'a> {
     cache: StoreCache,
 }
 
+/// Asynchronously writes set operations to `cold_db` in batches.
+/// Collects operation in batch, until its size is bigger than `min_transaction_size`.
+/// Then creates a thread to write that information to `cold_db`.
+/// Then joins some old thread,
+/// until size of transactions in `handles` is smaller than `max_handles_size`.
+/// Takes care of writing information to `cold_db`,
+/// [`write`] and [`join`] for all handles will be called on destruction.
+struct BatchTransaction<D: Database + 'static> {
```

Comment:
How about intro like this:
The BatchTransaction can be used to write multiple set operations to the cold db in batches. Each batch will be written in a dedicated thread.